### ISSUE TITLE
[tep-0093] Use Lister to fetch secret instead of making api requests

### ISSUE THREAD

**Issue by Yongxuanzhang** [tektoncd/pipeline#5884]
In https://github.com/tektoncd/pipeline/pull/5714 we fetch the secrets via api calls and this should be improved by using a lister. 
e.g.
https://github.com/sigstore/policy-controller/blob/ceeba7195243fc3dae3b48cfb7faf9fe6090f935/pkg/reconciler/clusterimagepolicy/controller.go#L69-L77

**Comment by vdemeester** [comment]
/remove-lifecycle stale
@Yongxuanzhang is it something we still need to tackle ?

**Comment by Yongxuanzhang** [comment]
> /remove-lifecycle stale @Yongxuanzhang is it something we still need to tackle ?

Yes, this issue will only affect trusted resources users so it is not high priority but I will address
